### PR TITLE
Add emfit_test.cpp for testing EMFit class

### PR DESCRIPTION
**Pull request by KimSangYeon-DGU**
Created `emfit_test.cpp` to verify the `EMFit` class.

### CODE REVIEW

**Comment by KimSangYeon-DGU** [comment]
I will increase the tolerance for covariance in the `EMFitEstimateTest`, because the bad result was detected in 12,736th loop.

**Comment by KimSangYeon-DGU** [comment]
Would it be better to add this test cases in `gmm_test.cpp` without creating a new file `emfit_test.cpp`?? And some test cases of GMM and HMM seem to be failed. I'll look into them to find robust cases.

File: src/mlpack/tests/emfit_test.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+  EMFit<> fitter;
+
+  // Check the default template parameters of EMFit class.
+  // The EMFit class's default clusterer and constraint for covariance are
+  // KMeans and PositiveDefiniteConstraint classes respectively.
+  if (std::is_same<decltype(fitter.Clusterer()), KMeans<>&>::value)
+    sameCluster = true;
+
+  if (std::is_same<decltype(fitter.Constraint()),
+    PositiveDefiniteConstraint&>::value)
```
Comment (rcurtin) [inline src/mlpack/tests/emfit_test.cpp]: This is a little pedantic but this should be indented twice. :+1:

File: src/mlpack/tests/emfit_test.cpp [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  // The EMFit class's default max-iteration is 300, and tolerance is 1e-10.
+  BOOST_CHECK_EQUAL(fitter.MaxIterations(), 300);
+  BOOST_CHECK_CLOSE(fitter.Tolerance(), 1e-10, 1e-3);
+  BOOST_CHECK_EQUAL(sameCluster, true);
+  BOOST_CHECK_EQUAL(sameConstraint, true);
+}
+
+/**
+ * Make sure EMFit can estimate parameters reasonably given observations
+ * from the Gaussian distributions.
```
Comment (rcurtin) [inline src/mlpack/tests/emfit_test.cpp]: Would you say that this is already tested via the `GMMTrainEMMultipleGaussians` test?

Comment (KimSangYeon-DGU) [reply src/mlpack/tests/emfit_test.cpp]: That makes sense. I thought it would be good to make some unit tests exclusively for `EMFit` class.

**Comment by KimSangYeon-DGU** [comment]
> But I'm not sure if the second test is kind of a duplicate of an existing test in `gmm_test.cpp`, and it looks like the first test is just checking that the default template types of `EMFit<>` are `KMeans<>` and `PositiveDefiniteContraint`?

Hi @rcurtin. That makes sense, I would like to make some unit test exclusively for `EMFit` class. but it seems to be unnecessary.

**Pull request closed without merging**